runebuf.go

```diff
@@ -449,11 +452,24 @@ func (r *runeBuffer) refresh(f func()) {
 		return
 	}
 
+	prevIdx := r.idx
+	prevBuf := append([]rune{}, r.buf...)
+
 	r.clean()
 	if f != nil {
 		f()
 	}
 	r.print()
+
+	if r.OnChange != nil {
+		if !runes.Equal(r.buf, prevBuf) {
+			if prevIdx != r.lastChangeIdx+1 {
```

Comment:
Is the idea of this check to stop `undo()` from writing its own undo entry?

Reply:
If i remember correctly it's to skip triggering change when just typing "normally" and not moving around so that we don't add undo entries for each character change

Reply:
maybe it would be cleaner to move that kind of logic into `opUndo` by passing some more arguments to OnChange?

Reply:
but i'm not very happy with the this whole undo implementation, feels a bit adhoc hacky, so if you have some idea how to implement it in a nicer way please do!